### PR TITLE
fix Buffer and Distributed engine parsing

### PR DESCRIPTION
**Pull request by phiSgr**
> [!NOTE]
> Refactors ClickHouse engine handling to wrap Buffer in BufferEngine, adds robust parse/serialize for Buffer/Distributed with nested optional rules and warnings, updates TS/Python generators and infra mapping, and adds comprehensive tests.
> 
> - **ClickHouse engine core**:
>   - **BufferEngine struct**: Introduces `BufferEngine` and changes `ClickhouseEngine::Buffer` to `Buffer(BufferEngine)`; centralizes building strings and nested optional flush params.
>   - **Parsing/Serialization**: Implements `Buffer(...)` and `Distributed(...)` parse/serialize with nested optional validation and warnings; updates `to_proto_string`, DDL generation, and `non_alterable_params_hash`.
>   - **Engine extraction**: Improves engine parsing from `CREATE TABLE`; logs warnings and records unsupported engines instead of defaulting silently.
>   - **Logging**: Adds `warn` usage for invalid param combinations.
>   - **Tests**: Adds round-trip and edge-case tests for Buffer/Distributed and invalid combinations.
> - **Codegen**:
>   - **Python/TypeScript**: Update generators to handle `ClickhouseEngine::Buffer(BufferEngine { ... })` and emit corresponding fields; import `BufferEngine`.
> - **Infra mapping**:
>   - Maps `EngineConfig::Buffer` to `ClickhouseEngine::Buffer(BufferEngine { ... })` during table conversion.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b349798b3e7a4d8ac340d6aca13594d7c7f9327a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/framework-cli/src/infrastructure/olap/clickhouse/mod.rs [hunk 1]
```diff
@@ -1881,13 +1881,30 @@ impl OlapOperations for ConfiguredDBClient {
...
                 // Try to parse the extracted engine definition
-                engine_def.as_str().try_into().ok()
+                match engine_def.as_str().try_into() {
+                    Ok(engine) => engine,
+                    Err(failed_str) => {
+                        warn!(
+                            "Failed to parse engine from CREATE TABLE query for table '{}': '{}'. This may indicate an unsupported engine type. Defaulting to MergeTree.",
+                            table_name, failed_str
+                        );
+                        ClickhouseEngine::MergeTree
```
Comment (onelesd) [inline apps/framework-cli/src/infrastructure/olap/clickhouse/mod.rs]: if i'm a user asking for a certain engine i'd prefer a failure
i may miss the warning and then go crazy trying to debug it later

Comment (phiSgr) [reply apps/framework-cli/src/infrastructure/olap/clickhouse/mod.rs]: so true.

File: apps/framework-cli/src/infrastructure/olap/clickhouse/queries.rs [hunk 2]
```diff
@@ -805,6 +810,119 @@ impl ClickhouseEngine {
...
+                min_bytes,
+                max_bytes,
+                flush_time,
+                flush_rows,
+                flush_bytes,
+            })
+        } else {
+            Err(original_value)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (phiSgr) [reply apps/framework-cli/src/infrastructure/olap/clickhouse/queries.rs:920]: WHAT
